editor/src/clj/editor/camera_editor.clj

```diff
@@ -273,30 +282,39 @@
                                           :aspect-ratio aspect-ratio
                                           :near-z near-z
                                           :far-z far-z
-                                          :is-orthographic orthographic-projection}
+                                          :is-orthographic orthographic-projection
+                                          :orthographic-zoom orthographic-zoom
+                                          :orthographic-mode orthographic-mode
+                                          :display-width project-display-width
+                                          :display-height project-display-height}
                               :passes [pass/outline]}}]}))
 
-(defn load-camera [_project self _resource camera-desc]
+(defn load-camera [project self _resource camera-desc]
   {:pre [(map? camera-desc)]} ; Camera$CameraDesc in map format.
-  (gu/set-properties-from-pb-map self Camera$CameraDesc camera-desc
-    aspect-ratio :aspect-ratio
-    fov :fov
-    near-z :near-z
-    far-z :far-z
-    auto-aspect-ratio (protobuf/int->boolean :auto-aspect-ratio)
-    orthographic-projection (protobuf/int->boolean :orthographic-projection)
-    orthographic-zoom :orthographic-zoom
-    orthographic-mode :orthographic-mode))
+  (concat
+    (g/connect project :display-width self :project-display-width)
+    (g/connect project :display-height self :project-display-height)
+    (gu/set-properties-from-pb-map self Camera$CameraDesc camera-desc
+      aspect-ratio :aspect-ratio
+      fov :fov
+      near-z :near-z
+      far-z :far-z
+      auto-aspect-ratio (protobuf/int->boolean :auto-aspect-ratio)
+      orthographic-projection (protobuf/int->boolean :orthographic-projection)
+      orthographic-zoom :orthographic-zoom
+      orthographic-mode :orthographic-mode)))
 
 (g/defnode CameraNode
   (inherits resource-node/ResourceNode)
 
   (property aspect-ratio g/Num ; Required protobuf field.
             (dynamic label (properties/label-dynamic :camera :aspect-ratio))
-            (dynamic tooltip (properties/tooltip-dynamic :camera :aspect-ratio)))
+            (dynamic tooltip (properties/tooltip-dynamic :camera :aspect-ratio))
+            (dynamic read-only? (g/fnk [orthographic-projection] orthographic-projection)))
```

Comment:
better UX for perspective/orthographic mode (show and hide fields depends on mode)